system/solaris_zone.py

```diff
@@ -0,0 +1,362 @@
+#!/usr/bin/python
+
+# (c) 2013, Paul Markham <pmarkham@netrefinery.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+import os
+import platform
+import tempfile
+
+DOCUMENTATION = '''
+---
+module: solaris_zone
+short_description: Manage Solaris zones
+description:
+   - Create, start, stop and delete Solaris zones. This module doesn't currently allow
+     changing of options for a zone that's already been created.
+version_added: "2.0"
+author: Paul Markham
+requirements:
+  - Solaris 10 or later
+options:
+  state:
+    required: true
+    description:
+      - C(present), create the zone.
+      - C(running), if the zone already exists, boot it, otherwise, create the zone
+          first, then boot it.
+      - C(started), synonym for C(running).
+      - C(stopped), shutdown a zone.
+      - C(absent), destroy the zone.
+    choices: ['present', 'started', 'running', 'stopped', 'absent']
+  name:
+    description:
+      - Zone name.
+    required: true
+  path:
+    description:
+      - The path where the zone will be created. This is required when the zone is created, but not
+        used otherwise.
+    required: false
+    default: null
+  sparse:
+    description:
+      - Whether to create a sparse (C(true)) or whole root (C(false)) zone.
+    required: false
+    default: false
+  root_password:
+    description:
+      - The password hash for the root account. If not specified, the zone's root account
+        will not have a password.
+    required: false
+    default: null
+  config:
+    required: false
+    description:
+      - 'The zonecfg configuration commands for this zone, separated by commas, e.g.
```

Comment:
Thinking about this a bit more, there's probably no reason this needs to be a list. It can probably just be a string so you use whatever is valid zonecfg syntax. I'll look at this as a future enhancement.